scaaml/capture/input_generators/attack_point_iterator.py

```diff
@@ -238,3 +241,53 @@ def __iter__(self) -> AttackPointIteratorT:
 
     def get_generated_keys(self) -> List[str]:
         return self._configuration_iterator.get_generated_keys()
+
+
+class AttackPointIteratorZip(AttackPointIterator):
+    """Attack point iterator zip class. This class takes any amount of operands
+    and combines them just like the `zip` function in Python."""
+
+    def __init__(self, operation: str, operands: List[Dict[str, Any]]) -> None:
+        """Initialize the zip iterator.
+          
+          Args:
+            operation (str): The operation of the iterator
+                represents what the iterator does and what 
+                has to be in the config file. This is only used once to
+                double check if the operation is the correct one.
+                
+            operands (List[Dict[str, Any]]): The operands are any number of
+                iterator configs that will be combined."""
+        assert "zip" == operation
+        self._operands = list(
+            build_attack_points_iterator(operand) for operand in operands)
+
+        list_of_lengths = [
+            operand._len for operand in self._operands if operand._len >= 0
+        ]
+        if not self._operands:
+            self._len = 0
+        else:
+            self._len = min(list_of_lengths, default=-1)
+
+    def __len__(self) -> int:
```

Comment:
NIT This implementation is repeated from the repeat iterator. It would also work for all others since we cache `self._len` and return it (there is an additional check for being non negative, but that is negligible). I would implement `__len__` only once on the base method.